### PR TITLE
Entra - Make Source field in payload required

### PR DESCRIPTION
**Pull request by hakimms**
This PR checks for source in the request payload and will return an error if not found.

[The Bug# 2161566](https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2161566)

### CODE REVIEW

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/CloudEventRequest.cs [hunk 1]
```diff
@@ -1,9 +1,10 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework.Validators;
+using System.ComponentModel.DataAnnotations;
 using System.Net.Http;
 using System.Text.Json.Serialization;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework.Validators;
```
[the remark this answers is not in the document]
Comment (hakimms) [reply sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/CloudEventRequest.cs:7]: Yes, this class needs to be public so it's properties and the TokenIssuanceStart class that inherits CloudEventRequest can be accessed by the customer. 
image

**Review by jsquire — APPROVED** [review]

**Comment by jsquire** [comment]
@hakimms:  Please let me know when you'd like me to merge.

**Comment by jsquire** [comment]
@hakimms:  When you get a moment, please join the "Microsoft" and "Azure" GitHub organizations and ensure that they're set to public on your profile.

**Comment by hakimms** [comment]
> @hakimms: When you get a moment, please join the "Microsoft" and "Azure" GitHub organizations and ensure that they're set to public on your profile.

I see the option to follow these organizations, but how do I join them?

**Comment by hakimms** [comment]
Please go ahead and merge this. Thank you

**Comment by hakimms** [comment]
> @hakimms: Please let me know when you'd like me to merge.

Please go ahead and merge it, thank you!

**Comment by jsquire** [comment]
> > @hakimms: When you get a moment, please join the "Microsoft" and "Azure" GitHub organizations and ensure that they're set to public on your profile.
> 
> I see the option to follow these organizations, but how do I join them?

https://docs.opensource.microsoft.com/github/opensource/accounts/linking/https://docs.opensource.microsoft.com/github/opensource/accounts/linking/

**Comment by jsquire** [comment]
Missed a URL that may be a secret leak.   Please evaluate and roll keys if needed:
https://github.com/Azure/azure-sdk-for-net/security/secret-scanning/181

**Pull request merged**